pkg/products/threescale/reconciler.go

```diff
@@ -2411,3 +2022,203 @@ func (r *Reconciler) ensureDeploymentConfigsReady(ctx context.Context, serverCli
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, err
+	}
+
+	// rate limit cluster
+	ratelimitClusterResource := ratelimit.CreateClusterResource(
+		ratelimitServiceCR.Spec.ClusterIP,
+		ratelimit.RateLimitClusterName,
+		getRatelimitServicePort(ratelimitServiceCR),
+	)
+	ratelimitClusterResource.Http2ProtocolOptions = &envoycore.Http2ProtocolOptions{}
+
+	// apicast cluster
+	apiCastClusterResource := ratelimit.CreateClusterResource(
+		ApicastContainerAddress,
+		ApicastClusterName,
+		ApicastContainerPort,
+	)
+
+	// apicast listener
+	apiCastFilters, _ := getListenerResourceFilters(
+		getAPICastVirtualHosts(ApicastClusterName),
+		getAPICastHTTPFilters(),
+	)
+	apiCastListenerResource := ratelimit.CreateListenerResource(
+		ApicastListenerName,
+		ApicastEnvoyProxyAddress,
+		ApicastEnvoyProxyPort,
+		apiCastFilters,
+	)
+
+	// create envoy config for apicast
+	apiCastProxyConfig := ratelimit.NewEnvoyConfig(ApicastClusterName, r.Config.GetNamespace(), ApicastNodeID)
+	err = apiCastProxyConfig.CreateEnvoyConfig(ctx, serverClient, []*envoyapi.Cluster{apiCastClusterResource, ratelimitClusterResource}, []*envoyapi.Listener{apiCastListenerResource}, installation)
+	if err != nil {
+		r.log.Errorf("Failed to create envoyconfig for apicast", l.Fields{"APICast": ApicastClusterName}, err)
+		return integreatlyv1alpha1.PhaseFailed, err
+	}
+
+	// backend-listener cluster
+	backendClusterResource := ratelimit.CreateClusterResource(
+		BackendContainerAddress,
+		BackendClusterName,
+		BackendContainerPort,
+	)
+
+	backendHTTPFilters, _ := getBackendListenerHTTPFilters()
+	// backend listener listener
+	backendFilters, _ := getListenerResourceFilters(
+		getBackendListenerVitualHosts(BackendClusterName),
+		backendHTTPFilters,
+	)
+	backendListenerResource := ratelimit.CreateListenerResource(
+		BackendListenerName,
+		BackendEnvoyProxyAddress,
+		BackendEnvoyProxyPort,
+		backendFilters,
+	)
+
+	// create envoy config for apicast
+	backendProxyConfig := ratelimit.NewEnvoyConfig(BackendClusterName, r.Config.GetNamespace(), BackendNodeID)
+	err = backendProxyConfig.CreateEnvoyConfig(ctx, serverClient, []*envoyapi.Cluster{backendClusterResource, ratelimitClusterResource}, []*envoyapi.Listener{backendListenerResource}, installation)
+	if err != nil {
+		r.log.Errorf("Failed to create envoyconfig for backend-listener", l.Fields{"BackendListener": BackendClusterName}, err)
+		return integreatlyv1alpha1.PhaseFailed, err
+	}
+
+	return integreatlyv1alpha1.PhaseCompleted, nil
+}
+
+func (r *Reconciler) getRateLimitServiceCR(ctx context.Context, serverClient k8sclient.Client) (*corev1.Service, error) {
+	rateLimitService := &corev1.Service{}
+	marin3rConfig, err := r.ConfigManager.ReadMarin3r()
+	if err != nil {
+		return nil, fmt.Errorf("failed to load marin3r config in 3scale reconciler: %v", err)
+	}
+	err = serverClient.Get(ctx, k8sclient.ObjectKey{
+		Namespace: marin3rConfig.GetNamespace(),
+		Name:      "ratelimit",
```

Comment:
only a single usage, but still a hardcoded value. Not sure should be changed to a constant/var tho :thinking:

Reply:
this route is created by the 3scale operator, so I am not sure if there is a constant with the service name there.